common/environment/environment.cpp

```diff
@@ -31,11 +31,75 @@
 #include "dalienv.hpp"
 
 #define SDS_LOCK_TIMEOUT  30000
-
+#define DEFAULT_DROPZONE_INDEX      1
+#define DROPZONE_BY_MACHINE_SUFFIX  "-dropzoneByMachine-"
+#define DROPZONE_SUFFIX             "dropzone-"
+#define MACHINE_PREFIX              "machine-"
+#define NUMBER_OF_DROPZONES_SUFFIX  "-numberOfDropZones"
 
 static int environmentTraceLevel = 1;
 static Owned <IConstEnvironment> cache;
 
+class CLocalEnvironment;
+
+class CConstMachineInfoIterator : public CInterface, implements IConstMachineInfoIterator
+{
+public:
+    CConstMachineInfoIterator();
+    IMPLEMENT_IINTERFACE
+    virtual bool first();
+    virtual bool next();
+    virtual bool isValid();
+    virtual IConstMachineInfo & query();
+    virtual unsigned count();
+
+protected:
+    IConstMachineInfo * curr;
+    Owned<CLocalEnvironment> constEnv;
+    unsigned index;
+    unsigned maxIndex;
```

Comment:
minor: Should we always be trying to make use of c++11 member initializations now, e.g. here